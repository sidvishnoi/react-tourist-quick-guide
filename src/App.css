* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background: gold;
}

.Content {
  padding: 1em;
}

h2.source {
  padding: 1em 1em 0;
  text-align: center;
}
