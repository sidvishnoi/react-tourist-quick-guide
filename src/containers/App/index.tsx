import * as React from 'react';
import { connect } from 'react-redux';

import { addCity, getSuggestions, selectLocation } from '../../actions';
import { State } from '../../state';

import Text from '../../components/Text';
import AutoComplete from '../AutoComplete';
import Cities from '../Cities';

const mapStateToProps = (state: State) => ({
  source: state.source,
});

function App({ source }: { source: string }) {
  if (!source) {
    return (
      <div className="App">
        <Text as="h1" size="fsize_09">
          Where are you travelling from?
        </Text>
        <AutoComplete
          placeholder="Your Location"
          onChange={getSuggestions}
          onSelect={selectLocation}
        />
      </div>
    );
  }

  return (
    <div className="App">
      <Text as="h1" size="fsize_09">
        Destinations from <em>{source}</em>:
      </Text>
      <AutoComplete
        placeholder="Add destination"
        onChange={getSuggestions}
        onSelect={addCity}
      />
      <div className="Content">
        <Cities />
      </div>
    </div>
  );
}

export default connect(mapStateToProps)(App);
