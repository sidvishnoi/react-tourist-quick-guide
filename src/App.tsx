import * as React from 'react';
import { connect } from 'react-redux';

import { addCity, getSuggestions, selectLocation } from './actions';
import './App.css';
import AutoCompleteSearch from './containers/AutoCompleteSearch';
import Cities from './containers/Cities';
import { State } from './state';

const mapStateToProps = ({ source }: State) => ({
  source,
});

function App({ source }: { source: string }) {
  if (!source) {
    return (
      <div className="App">
        <div className="Search">
          <AutoCompleteSearch
            placeholder="Your Location"
            onChange={getSuggestions}
            onSelect={selectLocation}
          />
        </div>
      </div>
    );
  }

  return (
    <div className="App">
      <h2 className="source">
        Destinations from <em>{source}</em>:
      </h2>
      <div className="Search">
        <AutoCompleteSearch
          placeholder="Add destination"
          onChange={getSuggestions}
          onSelect={addCity}
        />
      </div>
      <div className="Content">
        <Cities />
      </div>
    </div>
  );
}

export default connect(mapStateToProps)(App);
