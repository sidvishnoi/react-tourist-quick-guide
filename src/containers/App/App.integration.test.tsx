import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import { Provider } from 'react-redux';
import * as renderer from 'react-test-renderer';
import { mocked } from 'ts-jest/utils';

import App from '.';
import configureStore from '../../configureStore';
import initialState, { State } from '../../state';

import distanceAPI from '../../apis/distance';
import locationAPI from '../../apis/location';
import placesAPI from '../../apis/places';
import weatherAPI from '../../apis/weather';

jest.mock('../../apis/distance');
jest.mock('../../apis/location');
jest.mock('../../apis/places');
jest.mock('../../apis/weather');

Enzyme.configure({ adapter: new Adapter() });

const mockResponses = {
  distance: 100,
  weather: {
    forecast: [
      {
        icon: 113,
        summary: 'sunny',
        temperature: 30,
      },
      {
        icon: 230,
        summary: 'sunny',
        temperature: 35,
      },
      {
        icon: 263,
        summary: 'sunny',
        temperature: 32,
      },
    ],
    icon: 227,
    summary: 'sunny',
    temperature: 30,
  },
  location: [
    [{ id: '1', name: 'LOCATION' }],
    [{ id: '2', name: 'LOCATION TWO' }],
    [{ id: '3', name: 'LOCATION THREE' }],
  ],
  places: [
    { title: 'ONE', link: '#', summary: 'summary for ONE' },
    { title: 'TWO', link: '#', summary: 'summary for TWO' },
    { title: 'THREE', link: '#', summary: 'summary for THREE' },
    { title: 'FOUR', link: '#', summary: 'summary for FOUR' },
  ],
};

mocked(distanceAPI).mockResolvedValue(mockResponses.distance);
mocked(locationAPI)
  .mockResolvedValueOnce(mockResponses.location[0])
  .mockResolvedValueOnce(mockResponses.location[1])
  .mockResolvedValue(mockResponses.location[2]);
mocked(placesAPI).mockResolvedValue(mockResponses.places);
mocked(weatherAPI).mockResolvedValue(mockResponses.weather);

const delay = (duration: number) =>
  new Promise(resolve => setTimeout(resolve, duration));

describe('App', () => {
  const createApp = (state: State) => {
    const store = configureStore(state);
    const ReactApp = (
      <Provider store={store}>
        <App />
      </Provider>
    );
    const app = mount(ReactApp);
    return { store, app, ReactApp };
  };

  const readyState = {
    ...initialState,
    source: 'LOCATION',
    cities: {
      allIds: ['LOCATION TWO', 'LOCATION THREE'],
      byId: {
        'LOCATION TWO': {
          weather: { state: 'ready', data: mockResponses.weather },
          places: { state: 'ready', data: mockResponses.places },
          distance: { state: 'ready', data: mockResponses.distance },
        },
        'LOCATION THREE': {
          weather: { state: 'ready', data: mockResponses.weather },
          places: { state: 'ready', data: mockResponses.places },
          distance: { state: 'ready', data: mockResponses.distance },
        },
      },
    },
  };

  it('lets add a source city', async () => {
    const { app, store } = createApp(initialState);

    expect(app.find('h1').text()).toEqual('Where are you travelling from?');

    const input = app.find('input');
    input.simulate('change', { target: { value: 'FOO' } });
    await delay(310);
    input.simulate('keyDown', { key: 'ArrowDown', keyCode: 40, which: 40 });

    expect(store.getState().search.query).toEqual('FOO');
    expect(store.getState().search.suggestions.length).toEqual(1);
    const menu = app.find('div[role="listbox"]');
    const options = menu.find('div[role="option"]');
    expect(options.length).toEqual(1);

    expect(store.getState().source).toEqual('');
    options.at(0).simulate('click');
    expect(store.getState().source).toEqual('LOCATION');
    expect(app.find('h1').text()).toEqual('Destinations from LOCATION:');
  });

  it('lets add a new destination', async () => {
    const { app, store, ReactApp } = createApp({
      ...initialState,
      source: 'LOCATION',
    });
    expect(app.find('h1').text()).toEqual('Destinations from LOCATION:');
    expect(store.getState().source).toEqual('LOCATION');

    const input = app.find('input');
    input.simulate('change', { target: { value: 'FOO' } });
    await delay(310);
    input.simulate('keyDown', { key: 'ArrowDown', keyCode: 40, which: 40 });

    let options = app.find('div[role="listbox"] div[role="option"]');
    expect(options.length).toEqual(1);

    expect(store.getState().cities.allIds.length).toEqual(0);

    options.at(0).simulate('click');
    await delay(10);
    let state = store.getState().cities;
    expect(state.allIds).toEqual(['LOCATION TWO']);
    let city = state.byId['LOCATION TWO'];
    expect(city).toBeDefined();
    expect(city.weather.data).toEqual(mockResponses.weather);
    expect(city.places.data).toEqual(mockResponses.places);
    expect(city.distance.data).toEqual(mockResponses.distance);

    expect(renderer.create(ReactApp)).toMatchSnapshot();

    input.simulate('change', { target: { value: 'BAR' } });
    await delay(310);
    input.simulate('keyDown', { key: 'ArrowDown', keyCode: 40, which: 40 });
    options = app.find('div[role="listbox"] div[role="option"]');
    options.at(0).simulate('click');
    await delay(10);
    state = store.getState().cities;
    expect(state.allIds).toEqual(['LOCATION THREE', 'LOCATION TWO']);
    city = state.byId['LOCATION THREE'];
    expect(city).toBeDefined();
    expect(city.weather.data).toEqual(mockResponses.weather);
    expect(city.places.data).toEqual(mockResponses.places);
    expect(city.distance.data).toEqual(mockResponses.distance);

    expect(renderer.create(ReactApp)).toMatchSnapshot();
  });

  it('lets move a city up in destinations list', () => {
    const { app, store, ReactApp } = createApp(readyState);

    expect(app.find('div.city').length).toEqual(2);
    const moveButton = app
      .find('div.city')
      .at(1)
      .find('button.move');

    expect(store.getState().cities.allIds).toEqual([
      'LOCATION TWO',
      'LOCATION THREE',
    ]);
    moveButton.simulate('click');
    expect(store.getState().cities.allIds).toEqual([
      'LOCATION THREE',
      'LOCATION TWO',
    ]);
    expect(renderer.create(ReactApp)).toMatchSnapshot();
  });

  it('lets remove a city from destinations list', () => {
    const { app, store, ReactApp } = createApp(readyState);

    expect(app.find('div.city').length).toEqual(2);
    const removeButton = app
      .find('div.city')
      .first()
      .find('button.remove');

    expect(store.getState().cities.allIds).toEqual([
      'LOCATION TWO',
      'LOCATION THREE',
    ]);
    removeButton.simulate('click');
    expect(store.getState().cities.allIds).toEqual([
      'LOCATION THREE',
    ]);
    expect(renderer.create(ReactApp)).toMatchSnapshot();
  });
});
