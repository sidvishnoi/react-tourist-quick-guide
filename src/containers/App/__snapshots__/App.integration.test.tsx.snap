// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App lets add a new destination 1`] = `
<div
  className="App"
>
  <h1
    className="sc-bdVaJa fCEYOb"
    color="#000"
    size="fsize_09"
  >
    Destinations from 
    <em>
      LOCATION
    </em>
    :
  </h1>
  <div
    className="sc-bwzfXH fQaCWc"
  >
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <input
        aria-autocomplete="list"
        aria-expanded={false}
        autoComplete="off"
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add destination"
        role="combobox"
        value=""
      />
    </div>
  </div>
  <div
    className="Content"
  >
    <div
      className="city sc-cSHVUG hpSfXF"
    >
      <div
        className="sc-bZQynM lkaWTh"
      >
        <div
          className="buttons"
        >
          <button
            className="move sc-EHOje caNcft"
            color="#fff"
            onClick={[Function]}
            title="Move destination up in list"
          >
            ⬆
          </button>
          <button
            className="remove sc-EHOje hWrgYh"
            color="#fff"
            onClick={[Function]}
            title="Remove from list"
          >
            ✗
          </button>
        </div>
        <h2
          className="sc-bdVaJa sc-gzVnrw gmWgic"
          color="#fff"
          size="fsize_08"
        >
          LOCATION TWO
        </h2>
        <span
          className="sc-bdVaJa sc-htoDjs jyuKTm"
          color="#fff"
          size="fsize_04"
        >
          100 KM
        </span>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div
          className="sc-VigVT hpjzoH"
        >
          <div
            className="weather-today"
          >
            <div
              className="weather-day sc-gqjmRU jFBVhn"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/227.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
          <div
            className="weather-forecast sc-jTzLTM kAcagK"
          >
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/113.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/230.png)",
                }
              }
            >
              <span>
                35
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/263.png)",
                }
              }
            >
              <span>
                32
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div>
          <h3
            className="sc-bdVaJa ldjkJc"
            color="#000"
            size="fsize_05"
          >
            Top tourist places:
          </h3>
          <ul
            className="sc-gZMcBi fRKiLq"
          >
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  ONE
                </a>
                :
                 
                <em>
                  summary for ONE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  TWO
                </a>
                :
                 
                <em>
                  summary for TWO
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  THREE
                </a>
                :
                 
                <em>
                  summary for THREE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  FOUR
                </a>
                :
                 
                <em>
                  summary for FOUR
                </em>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX ulTBn"
    >
      Advertisement Space
    </div>
  </div>
</div>
`;

exports[`App lets add a new destination 2`] = `
<div
  className="App"
>
  <h1
    className="sc-bdVaJa fCEYOb"
    color="#000"
    size="fsize_09"
  >
    Destinations from 
    <em>
      LOCATION
    </em>
    :
  </h1>
  <div
    className="sc-bwzfXH fQaCWc"
  >
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <input
        aria-autocomplete="list"
        aria-expanded={false}
        autoComplete="off"
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add destination"
        role="combobox"
        value=""
      />
    </div>
  </div>
  <div
    className="Content"
  >
    <div
      className="city sc-cSHVUG hpSfXF"
    >
      <div
        className="sc-bZQynM lkaWTh"
      >
        <div
          className="buttons"
        >
          <button
            className="move sc-EHOje caNcft"
            color="#fff"
            onClick={[Function]}
            title="Move destination up in list"
          >
            ⬆
          </button>
          <button
            className="remove sc-EHOje hWrgYh"
            color="#fff"
            onClick={[Function]}
            title="Remove from list"
          >
            ✗
          </button>
        </div>
        <h2
          className="sc-bdVaJa sc-gzVnrw gmWgic"
          color="#fff"
          size="fsize_08"
        >
          LOCATION THREE
        </h2>
        <span
          className="sc-bdVaJa sc-htoDjs jyuKTm"
          color="#fff"
          size="fsize_04"
        >
          100 KM
        </span>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div
          className="sc-VigVT hpjzoH"
        >
          <div
            className="weather-today"
          >
            <div
              className="weather-day sc-gqjmRU jFBVhn"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/227.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
          <div
            className="weather-forecast sc-jTzLTM kAcagK"
          >
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/113.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/230.png)",
                }
              }
            >
              <span>
                35
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/263.png)",
                }
              }
            >
              <span>
                32
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div>
          <h3
            className="sc-bdVaJa ldjkJc"
            color="#000"
            size="fsize_05"
          >
            Top tourist places:
          </h3>
          <ul
            className="sc-gZMcBi fRKiLq"
          >
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  ONE
                </a>
                :
                 
                <em>
                  summary for ONE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  TWO
                </a>
                :
                 
                <em>
                  summary for TWO
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  THREE
                </a>
                :
                 
                <em>
                  summary for THREE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  FOUR
                </a>
                :
                 
                <em>
                  summary for FOUR
                </em>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX ulTBn"
    >
      Advertisement Space
    </div>
    <div
      className="city sc-cSHVUG hpSfXF"
    >
      <div
        className="sc-bZQynM lkaWTh"
      >
        <div
          className="buttons"
        >
          <button
            className="move sc-EHOje caNcft"
            color="#fff"
            onClick={[Function]}
            title="Move destination up in list"
          >
            ⬆
          </button>
          <button
            className="remove sc-EHOje hWrgYh"
            color="#fff"
            onClick={[Function]}
            title="Remove from list"
          >
            ✗
          </button>
        </div>
        <h2
          className="sc-bdVaJa sc-gzVnrw gmWgic"
          color="#fff"
          size="fsize_08"
        >
          LOCATION TWO
        </h2>
        <span
          className="sc-bdVaJa sc-htoDjs jyuKTm"
          color="#fff"
          size="fsize_04"
        >
          100 KM
        </span>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div
          className="sc-VigVT hpjzoH"
        >
          <div
            className="weather-today"
          >
            <div
              className="weather-day sc-gqjmRU jFBVhn"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/227.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
          <div
            className="weather-forecast sc-jTzLTM kAcagK"
          >
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/113.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/230.png)",
                }
              }
            >
              <span>
                35
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/263.png)",
                }
              }
            >
              <span>
                32
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div>
          <h3
            className="sc-bdVaJa ldjkJc"
            color="#000"
            size="fsize_05"
          >
            Top tourist places:
          </h3>
          <ul
            className="sc-gZMcBi fRKiLq"
          >
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  ONE
                </a>
                :
                 
                <em>
                  summary for ONE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  TWO
                </a>
                :
                 
                <em>
                  summary for TWO
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  THREE
                </a>
                :
                 
                <em>
                  summary for THREE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  FOUR
                </a>
                :
                 
                <em>
                  summary for FOUR
                </em>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App lets move a city up in destinations list 1`] = `
<div
  className="App"
>
  <h1
    className="sc-bdVaJa fCEYOb"
    color="#000"
    size="fsize_09"
  >
    Destinations from 
    <em>
      LOCATION
    </em>
    :
  </h1>
  <div
    className="sc-bwzfXH fQaCWc"
  >
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <input
        aria-autocomplete="list"
        aria-expanded={false}
        autoComplete="off"
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add destination"
        role="combobox"
        value=""
      />
    </div>
  </div>
  <div
    className="Content"
  >
    <div
      className="city sc-cSHVUG hpSfXF"
    >
      <div
        className="sc-bZQynM lkaWTh"
      >
        <div
          className="buttons"
        >
          <button
            className="move sc-EHOje caNcft"
            color="#fff"
            onClick={[Function]}
            title="Move destination up in list"
          >
            ⬆
          </button>
          <button
            className="remove sc-EHOje hWrgYh"
            color="#fff"
            onClick={[Function]}
            title="Remove from list"
          >
            ✗
          </button>
        </div>
        <h2
          className="sc-bdVaJa sc-gzVnrw gmWgic"
          color="#fff"
          size="fsize_08"
        >
          LOCATION THREE
        </h2>
        <span
          className="sc-bdVaJa sc-htoDjs jyuKTm"
          color="#fff"
          size="fsize_04"
        >
          100 KM
        </span>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div
          className="sc-VigVT hpjzoH"
        >
          <div
            className="weather-today"
          >
            <div
              className="weather-day sc-gqjmRU jFBVhn"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/227.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
          <div
            className="weather-forecast sc-jTzLTM kAcagK"
          >
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/113.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/230.png)",
                }
              }
            >
              <span>
                35
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/263.png)",
                }
              }
            >
              <span>
                32
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div>
          <h3
            className="sc-bdVaJa ldjkJc"
            color="#000"
            size="fsize_05"
          >
            Top tourist places:
          </h3>
          <ul
            className="sc-gZMcBi fRKiLq"
          >
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  ONE
                </a>
                :
                 
                <em>
                  summary for ONE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  TWO
                </a>
                :
                 
                <em>
                  summary for TWO
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  THREE
                </a>
                :
                 
                <em>
                  summary for THREE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  FOUR
                </a>
                :
                 
                <em>
                  summary for FOUR
                </em>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX ulTBn"
    >
      Advertisement Space
    </div>
    <div
      className="city sc-cSHVUG hpSfXF"
    >
      <div
        className="sc-bZQynM lkaWTh"
      >
        <div
          className="buttons"
        >
          <button
            className="move sc-EHOje caNcft"
            color="#fff"
            onClick={[Function]}
            title="Move destination up in list"
          >
            ⬆
          </button>
          <button
            className="remove sc-EHOje hWrgYh"
            color="#fff"
            onClick={[Function]}
            title="Remove from list"
          >
            ✗
          </button>
        </div>
        <h2
          className="sc-bdVaJa sc-gzVnrw gmWgic"
          color="#fff"
          size="fsize_08"
        >
          LOCATION TWO
        </h2>
        <span
          className="sc-bdVaJa sc-htoDjs jyuKTm"
          color="#fff"
          size="fsize_04"
        >
          100 KM
        </span>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div
          className="sc-VigVT hpjzoH"
        >
          <div
            className="weather-today"
          >
            <div
              className="weather-day sc-gqjmRU jFBVhn"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/227.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
          <div
            className="weather-forecast sc-jTzLTM kAcagK"
          >
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/113.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/230.png)",
                }
              }
            >
              <span>
                35
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/263.png)",
                }
              }
            >
              <span>
                32
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div>
          <h3
            className="sc-bdVaJa ldjkJc"
            color="#000"
            size="fsize_05"
          >
            Top tourist places:
          </h3>
          <ul
            className="sc-gZMcBi fRKiLq"
          >
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  ONE
                </a>
                :
                 
                <em>
                  summary for ONE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  TWO
                </a>
                :
                 
                <em>
                  summary for TWO
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  THREE
                </a>
                :
                 
                <em>
                  summary for THREE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  FOUR
                </a>
                :
                 
                <em>
                  summary for FOUR
                </em>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App lets remove a city from destinations list 1`] = `
<div
  className="App"
>
  <h1
    className="sc-bdVaJa fCEYOb"
    color="#000"
    size="fsize_09"
  >
    Destinations from 
    <em>
      LOCATION
    </em>
    :
  </h1>
  <div
    className="sc-bwzfXH fQaCWc"
  >
    <div
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <input
        aria-autocomplete="list"
        aria-expanded={false}
        autoComplete="off"
        className=""
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add destination"
        role="combobox"
        value=""
      />
    </div>
  </div>
  <div
    className="Content"
  >
    <div
      className="city sc-cSHVUG hpSfXF"
    >
      <div
        className="sc-bZQynM lkaWTh"
      >
        <div
          className="buttons"
        >
          <button
            className="move sc-EHOje caNcft"
            color="#fff"
            onClick={[Function]}
            title="Move destination up in list"
          >
            ⬆
          </button>
          <button
            className="remove sc-EHOje hWrgYh"
            color="#fff"
            onClick={[Function]}
            title="Remove from list"
          >
            ✗
          </button>
        </div>
        <h2
          className="sc-bdVaJa sc-gzVnrw gmWgic"
          color="#fff"
          size="fsize_08"
        >
          LOCATION THREE
        </h2>
        <span
          className="sc-bdVaJa sc-htoDjs jyuKTm"
          color="#fff"
          size="fsize_04"
        >
          100 KM
        </span>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div
          className="sc-VigVT hpjzoH"
        >
          <div
            className="weather-today"
          >
            <div
              className="weather-day sc-gqjmRU jFBVhn"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/227.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
          <div
            className="weather-forecast sc-jTzLTM kAcagK"
          >
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/113.png)",
                }
              }
            >
              <span>
                30
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/230.png)",
                }
              }
            >
              <span>
                35
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
            <div
              className="weather-day sc-gqjmRU libNPB"
              style={
                Object {
                  "backgroundImage": "url(/images/weather-icons/263.png)",
                }
              }
            >
              <span>
                32
              </span>
              °C
              <p>
                sunny
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-fjdhpX laTUCL"
      >
        <div>
          <h3
            className="sc-bdVaJa ldjkJc"
            color="#000"
            size="fsize_05"
          >
            Top tourist places:
          </h3>
          <ul
            className="sc-gZMcBi fRKiLq"
          >
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  ONE
                </a>
                :
                 
                <em>
                  summary for ONE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  TWO
                </a>
                :
                 
                <em>
                  summary for TWO
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  THREE
                </a>
                :
                 
                <em>
                  summary for THREE
                </em>
              </span>
            </li>
            <li
              className="sc-iwsKbI cHNZZb"
            >
              <span>
                <a
                  className="sc-dnqmqq cCpFAY"
                  href="#"
                >
                  FOUR
                </a>
                :
                 
                <em>
                  summary for FOUR
                </em>
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="sc-ifAKCX ulTBn"
    >
      Advertisement Space
    </div>
  </div>
</div>
`;
